fix: imported optional Map type bindings in Rust

krisbitney opened the pull request:
There are two issues affecting imported Map type bindings from compiling in Rust.

**First**
```rust
// expected type option, got map
let mut _headers: Option<Map<String, String>> = Map::<String, String>::new();
let mut _url_params: Option<Map<String, String>> = Map::<String, String>::new();
```

**Second**
In the object serialization logic for optional Map type, we are trying to pass a String instead of a Result<String, Error>. Example:
```
"headers" => {
                reader.context().push(&field, "Option<Map<String, String>>", "type found, reading property");
                _headers = reader.read_optional_ext_generic_map(|reader| {
                    reader.read_string()? //  <---- This `?` should not be here!
                }, |reader| {
                    reader.read_string()
                })?; 
                reader.context().pop();
            }
```

krisbitney committed:
fix for bindings for imported optional Rust Map type

cbrzn commented:
nice catch ser :pray:, I previously fixed this but only in object types (https://github.com/polywrap/monorepo/pull/1142) and not for imported object types. I guess you also need to update `schema/bind` test and we are ready to go

dOrgJelli reviewed (dismissed):
Nice catch! Just need tests to go green

krisbitney committed:
updated rs schema binding tests

krisbitney commented:
> nice catch ser 🙏, I previously fixed this but only in object types (#1142) and not for imported object types. I guess you also need to update `schema/bind` test and we are ready to go

Thanks. 😄

krisbitney commented:
> Nice catch! Just need tests to go green

Thanks! 😄

Niraj-Kamdar reviewed (approved):
@krisbitney Is it also fixed for the methodTypes?

krisbitney commented:
> @krisbitney Is it also fixed for the methodTypes?

I fixed both the issues mentioned in the PR description. Are you referring to one of those or something else?

cbrzn commented:
> > @krisbitney Is it also fixed for the methodTypes?
> 
> I fixed both the issues mentioned in the PR description. Are you referring to one of those or something else?

Just checked it and we are good to go ser
